tool/typecheck.ts

```diff
@@ -198,6 +207,13 @@ class TypecheckStream extends Stream.Transform {
                 schemaRetriever: this._schemas,
             }, true);
 
+            const preserved_tokens : string[] = Object.keys(EntityUtils.ENTITIES);
+            preserved_tokens.push('GENERIC_ENTITY');
+            const preserved_regex = new RegExp(preserved_tokens.join("|"));
+
+            if (!this._tokenized)
+                this._current.preprocessed = this._tokenizer.tokenize(this._current.preprocessed, preserved_regex).rawTokens.join(' ');
```

Comment:
What's going on here? The datasets that this tool processes are already tokenized.

Reply:
Don't you want to have the option though?
I found it useful to retokenize spl datasets where tokenization was done with previous code. More sentences typechecked as a result.

Reply:
That's not really a use case we want to support, no. Tokenizing an existing tokenized dataset is not guaranteed to be idempotent, or even do anything sensible.